core/templates/dev/head/components/forms/schema_editors/schema_based_list_editor_directive.html

```diff
@@ -25,7 +25,7 @@
 
   <!-- The margin-left is added here to left-align the button with the previous list items. -->
   <div style="height: 30px; margin-top: 4px;">
-    <button ng-show="!isOneLineInput || isAddItemButtonPresent"
+    <button ng-show="!isOneLineInput || isAddItemButtonPresent || (localValue.length && localValue[localValue.length - 1].length)"
```

Comment:
OK, thanks for explaining. One day we should refactor this (since isAddItemButtonPresent is a bit confusing if it isn't the only thing controlling this) but let's merge for now. Thanks!